### ISSUE TITLE
rootn.exe instant crash on startup

### ISSUE THREAD

**Issue by hansenjo** [root-project/root#14767]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

Running `rootn.exe` from ROOT 6.30.04 should bring up the ROOT prompt, but instead crashes instantly on startup. 

It works fine with older versions of ROOT (6.28, 6.26).

### Reproducer

- Install precompiled ROOT 6.30.04, downloaded from root.cern, on RHEL8 or 9.
- Run `thisroot.sh` to set up the runtime environment
- Type rootn.exe
- Hard crash

Transcript of session on RHEL8:
```
% source /apps/ROOT/6.30.04/bin/thisroot.sh
% root-config --version
6.30.04
% root-config --features
cxx17 asimage builtin_afterimage builtin_clang builtin_cling builtin_cppzmq builtin_davix builtin_ftgl builtin_gl2ps builtin_glew builtin_gtest builtin_llvm builtin_lz4 builtin_lzma builtin_nlohmannjson builtin_openui5 builtin_pcre builtin_tbb builtin_unuran builtin_vdt builtin_veccore builtin_xrootd builtin_xxhash builtin_zeromq builtin_zlib builtin_zstd clad dataframe davix fftw3 fitsio fortran gdml http imt mathmore mlp minuit2 mysql opengl pgsql proof pyroot roofit roofit_multiprocess webgui root7 rpath runtime_cxxmodules shadowpw shared soversion sqlite ssl tmva tmva-cpu tmva-pymva spectrum unuran vdt veccore x11 xml xrootd
% which rootn.exe
/apps/ROOT/6.30.04/bin/rootn.exe
% rootn.exe
Fatal in <operator new>: with std::align_val_t is not implemented yet
aborting
... longish stack trace ...
```
This works without error with ROOT 6.26.14 and 6.28.12 on the exact same machine. Running plain `root` 6.30.04 also works, of course. Only `rootn.exe` seems affected.

### ROOT version

 ```
  ------------------------------------------------------------------
  | Welcome to ROOT 6.30/04                        https://root.cern |
  | (c) 1995-2024, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Jan 31 2024, 09:46:18                 |
  | From heads/master@tags/v6-30-04                                  |
  | With c++ (GCC) 8.5.0 20210514 (Red Hat 8.5.0-21)                 |
  | Try '.help'/'.?', '.demo', '.license', '.credits', '.quit'/'.q'  |
   ------------------------------------------------------------------
```

### Installation method

pre-built binary, downloaded from root.cern

### Operating system

Linux RHEL8 or 9

### Additional context

Tried several different Linux hosts running RHEL8/CentOS 8, all with same result.

**Comment by ferdymercury** [comment]
Same issue on Ubuntu 22.04

**Comment by pcanal** [comment]
This fixed in [v6.32.00 ](https://github.com/root-project/root/pull/15343) and [v6.34.00 ](https://github.com/root-project/root/pull/14792)(master).
